src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -174,6 +179,74 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
+        private void CheckArgumentNullability(MethodSymbol attributeConstructor, AnalyzedAttributeArguments arguments, DiagnosticBag diagnostics)
+        {
+            // Check constructor arguments
+            for (int i = 0; i < attributeConstructor.ParameterCount && i < arguments.ConstructorArguments.Arguments.Count; i++)
+            {
+                var parameter = attributeConstructor.Parameters[i];
+                var argument = arguments.ConstructorArguments.Argument(i);
+                reportPossibleNullabilityMismatch(parameter.Type, argument);
+            }
+
+            // Check assignments
+            for (int i = 0; i < arguments.NamedArguments.Length; i++)
+            {
+                var assignment = arguments.NamedArguments[i] as BoundAssignmentOperator;
+                TypeSymbolWithAnnotations assignmentType;
+                switch (assignment.Left)
+                {
+                    case BoundPropertyAccess propertyAccess:
+                        assignmentType = propertyAccess.PropertySymbol.Type;
+                        break;
+                    case BoundFieldAccess fieldAccess:
+                        assignmentType = fieldAccess.FieldSymbol.Type;
+                        break;
+                    default:
+                        throw ExceptionUtilities.Unreachable;
+                }
+
+                reportPossibleNullabilityMismatch(assignmentType, assignment.Right);
+            }
+
+            void reportPossibleNullabilityMismatch(TypeSymbolWithAnnotations destination, BoundExpression sourceExpression)
+            {
+                if (IsNullableEnabled(sourceExpression.SyntaxTree, sourceExpression.Syntax.SpanStart))
```

Comment:
>if (IsNullableEnabled(sourceExpression.SyntaxTree, sourceExpression.Syntax.SpanStart)) [](start = 16, length = 86)

Would it make sense to do this earlier? Perhaps, at most once per attribute?

Reply:
Unfortunately we cant. A user can enable / disable nullable per argument, so we have to check it for each one [AttributeArgument_Constructor_NullLiteral_DisabledEnabled](https://github.com/dotnet/roslyn/pull/33816/files#diff-4ef499b331bacc63b10e31e95a00db61R81597) test has an example of this.